src/main/java/com/internet/shop/util/HashUtil.java

```diff
@@ -0,0 +1,32 @@
+package com.internet.shop.util;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import org.apache.log4j.Logger;
+
+public class HashUtil {
+    private static Logger logger = Logger.getLogger(HashUtil.class);
+
+    public static byte[] getSalt() {
+        SecureRandom random = new SecureRandom();
+        byte[] salt = new byte[16];
+        random.nextBytes(salt);
+        return salt;
+    }
+
+    public static String hashPassword(String password, byte[] salt) {
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA512");
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashedPassword.append(String.format("%2x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            logger.error(e);
```

Comment:
checklist says logging is not enough here. Programm must stop if there is an error in hashing

Reply:
agree let's throw an exception here